client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -69,59 +74,86 @@ function MembershipSubscriptions( {
 	return <MembershipsDataViews memberships={ memberships } />;
 }
 
-function isDataLoading( {
-	isFetchingUserPurchases,
+const PurchasesListDataView: React.FC<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps
+> = ( {
 	hasLoadedUserPurchasesFromServer,
-}: {
-	hasLoadedUserPurchasesFromServer: boolean;
-	isFetchingUserPurchases: boolean;
-} ) {
-	if ( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) {
-		return true;
-	}
-}
-
-function ConciergeBanner( {
+	isFetchingUserPurchases,
+	purchases,
+	subscriptions,
+	sites,
 	nextAppointment,
-	availableSessions,
 	isUserBlocked,
-}: Pick<
-	PurchasesListConnectedProps,
-	'nextAppointment' | 'availableSessions' | 'isUserBlocked'
-> ) {
-	return (
-		<PurchaseListConciergeBanner
-			nextAppointment={ nextAppointment ?? undefined }
-			availableSessions={ availableSessions }
-			isUserBlocked={ isUserBlocked }
-		/>
-	);
-}
+	availableSessions,
+	userId,
+} ) => {
+	const translate = useTranslate();
+	const {
+		data: transferredOwnershipPurchases = [],
+		isLoading,
+		isSuccess: hasLoadedTransferredOwnershipPurchases,
+	} = useGetJetpackTransferredLicensePurchases( { userId } );
+
+	const isDataLoading = useCallback( () => {
+		if (
+			( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) ||
+			( isLoading && ! hasLoadedTransferredOwnershipPurchases )
+		) {
+			return true;
+		}
+
+		return false;
+	}, [
+		hasLoadedUserPurchasesFromServer,
+		isFetchingUserPurchases,
+		isLoading,
+		hasLoadedTransferredOwnershipPurchases,
+	] );
+
+	const allPurchasesLoaded =
+		hasLoadedUserPurchasesFromServer && hasLoadedTransferredOwnershipPurchases;
+
+	const allPurchases = useMemo( () => {
+		if ( allPurchasesLoaded ) {
+			return [ ...purchases, ...transferredOwnershipPurchases ];
```

Comment:
Previous code assumed `purchases` might be null, so if the code crashes, it will be here, cause spread operator would run on undefined.